custom_components/mawaqit/sensor.py

```
@@ -1,34 +1,24 @@
-)
+from .const import DATA_UPDATED, DOMAIN, PRAYER_TIMES_ICON, SENSOR_TYPES
+
+import json
+import os
+import logging
 
 _LOGGER = logging.getLogger(__name__)
 
+def is_date_parsing(date_str):
+    try:
+        return bool(date_parser.parse(date_str))
```

Ruff (0.8.2)

Undefined name `date_parser`

(F821)